--- ./CcspPandM/custom/comcast/source/TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c
+++ ./CcspPandM/custom/comcast/source/TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c	2015-06-10 14:05:49.097030780 -0600
@@ -287,7 +287,23 @@
     } else {
         AnscTraceWarning(("%s: set WiFi ApplySetting OK\n", __FUNCTION__));
     }
-
+	//zqiu>>
+	if (g_SetParamValueBool("Device.X_COMCAST-COM_GRE.Tunnel.1.Enable", value) != ANSC_STATUS_SUCCESS) {
+        AnscTraceError(("%s: set X_COMCAST-COM_GRE.Tunnel.1.Enable error\n", __FUNCTION__));
+    } else {
+        AnscTraceWarning(("%s: set X_COMCAST-COM_GRE.Tunnel.1.Enable OK\n", __FUNCTION__));
+    }
+	if (g_SetParamValueBool("Device.X_COMCAST-COM_GRE.Tunnel.1.Interface.1.Enable", value) != ANSC_STATUS_SUCCESS) {
+        AnscTraceError(("%s: set X_COMCAST-COM_GRE.Tunnel.1.Interface.1.Enable error\n", __FUNCTION__));
+    } else {
+        AnscTraceWarning(("%s: set X_COMCAST-COM_GRE.Tunnel.1.Interface.1.Enable OK\n", __FUNCTION__));
+    }
+	if (g_SetParamValueBool("Device.X_COMCAST-COM_GRE.Tunnel.1.Interface.2.Enable", value) != ANSC_STATUS_SUCCESS) {
+        AnscTraceError(("%s: set X_COMCAST-COM_GRE.Tunnel.1.Interface.2.Enable error\n", __FUNCTION__));
+    } else {
+        AnscTraceWarning(("%s: set X_COMCAST-COM_GRE.Tunnel.1.Interface.2.Enable OK\n", __FUNCTION__));
+    }
+	//zqiu<<
     if (value)
         PsmSet(HOTSPOT_PSM_EANBLE, "1");
     else
