--- ./CcspPandM/source/TR-181/middle_layer_src/cosa_ethernet_dml.c
+++ ./CcspPandM/source/TR-181/middle_layer_src/cosa_ethernet_dml.c	2015-06-15 07:21:49.132341226 -0600
@@ -566,6 +566,17 @@
         return TRUE;
     }
 
+    if( AnscEqualString(ParamName, "CurrentBitRate", TRUE))
+    {
+        /* collect value */
+
+        CosaDmlEthPortGetDinfo(NULL, pEthernetPortFull->Cfg.InstanceNumber, &pEthernetPortFull->DynamicInfo);
+        
+        *puLong = pEthernetPortFull->DynamicInfo.CurrentBitRate;
+        return TRUE;
+    }
+
+
 
     /* CcspTraceWarning(("Unsupported parameter '%s'\n", ParamName)); */
     return FALSE;
