--- ./CcspPandM/source/TR-181/middle_layer_src/plugin_main_apis.h
+++ ./CcspPandM/source/TR-181/middle_layer_src/plugin_main_apis.h	2015-06-10 14:05:49.097030780 -0600
@@ -183,6 +183,7 @@
     ANSC_HANDLE                  hRLog;                                                     \
     ANSC_HANDLE                  hMultiLan;                                                 \
     ANSC_HANDLE                  hGRE;                                                      \
+    ANSC_HANDLE                  hTGRE;                                                     \
     ANSC_HANDLE                  hCGRE;                                                     \
     ANSC_HANDLE                  hHotspot;                                                  \
     ANSC_HANDLE                  hTSIP;                                                     \
