--- ./CcspPandM/source/TR-181/middle_layer_src/cosa_x_cisco_com_ddns_dml.c
+++ ./CcspPandM/source/TR-181/middle_layer_src/cosa_x_cisco_com_ddns_dml.c	2015-05-20 13:29:09.944854221 -0600
@@ -1360,7 +1360,7 @@
         }
         else
         {
-            AnscCopyString(pDdnsService->ServiceName, "None");
+			//AnscCopyString(pDdnsService->ServiceName, "None");
         }
 
         return TRUE;
