--- ./CcspPandM/custom/comcast/source/TR-181/custom_sbapi/cosa_x_comcast_com_gre_apis.c
+++ ./CcspPandM/custom/comcast/source/TR-181/custom_sbapi/cosa_x_comcast_com_gre_apis.c	2015-05-20 13:29:09.944854221 -0600
@@ -270,6 +270,9 @@
         
         inst = atoi(strrchr(curInt,'.')+1);
         
+		memset(paramname,0,sizeof(paramname));
+		memset(outdata,0,sizeof(outdata));
+        
         size = sizeof(outdata);
         
         if (syscfg_get(NULL, "wan_physical_ifname", paramname, sizeof(paramname)) != 0) {
@@ -290,14 +293,28 @@
             return -1;
         circuitSave = snprintf(circuitid, sizeof(circuitid), "%s;", varStruct.parameterValue);
 #endif
+        memset(paramname,0,sizeof(paramname));
+		memset(outdata,0,sizeof(outdata));
         
         size = sizeof(outdata);
         snprintf(paramname, sizeof(paramname),"%s.%s", curInt, "SSID");
         retval = COSAGetParamValueByPathName(bus_handle, &varStruct, &size);
         if ( retval != ANSC_STATUS_SUCCESS)
             return -1;
+
+		if(!(strcmp(varStruct.parameterValue,""))){
+			  snprintf(paramname, sizeof(paramname), "eRT.com.cisco.spvtg.ccsp.Device.WiFi.Radio.SSID.%d.SSID",inst);
+		  GrePsmGet(paramname,varStruct.parameterValue, size);
+		  if(strlen(varStruct.parameterValue)==0) {
+				strcpy(varStruct.parameterValue,"xfinitywifi");
+		  }
+		}
+        
         circuitSave += snprintf(circuitid + circuitSave, sizeof(circuitid) - circuitSave, "%s;", varStruct.parameterValue);
         
+	    memset(paramname,0,sizeof(paramname));
+	    memset(outdata,0,sizeof(outdata));
+        
         size = sizeof(outdata);
         snprintf(paramname, sizeof(paramname), "Device.WiFi.AccessPoint.%d.Security.ModeEnabled", inst);
         retval = COSAGetParamValueByPathName(bus_handle, &varStruct, &size);
@@ -401,6 +418,30 @@
     return 1;
 }
 
+#define kHotspotfd_tunnelEP                 "hotspotfd-tunnelEP" 
+
+ANSC_STATUS
+CosaDml_GreIfGetConnectedRemoteEndpoint(ULONG idx, COSA_DML_GRE_IF *greIf)
+{
+
+	char cmd[126] = {0};
+	char line_buf[126] = {0};
+	FILE *fp = NULL;
+
+	if(!greIf)
+		return ANSC_STATUS_FAILURE;
+
+	snprintf(cmd, sizeof(cmd), "sysevent get %s",kHotspotfd_tunnelEP);       
+    if (((fp = popen(cmd,"r")) != NULL) && (fgets(line_buf, sizeof(line_buf), fp)))
+    {
+		sprintf(greIf->ConnectedRemoteEndpoint,"%s",line_buf);
+    }
+	if(fp)
+		pclose(fp);
+
+	return ANSC_STATUS_SUCCESS;	
+}
+
 ANSC_STATUS
 CosaDml_GreIfGetEntryByIndex(ULONG idx, COSA_DML_GRE_IF *greIf)
 {
